apps/src/sites/studio/pages/components/editorRedux.js

```diff
@@ -0,0 +1,127 @@
+/* global levelKeyList */
+
+import _ from 'lodash';
+
+function updatePositions(node) {
+  for (var i = 0; i < node.length; i++) {
+    node[i].position = i + 1;
+  }
+}
+
+let newStageId = -1;
+let newLevelId = -1;
+
+export default function reducer(state, action) {
+  let newState = _.cloneDeep(state);
```

Comment:
This reducer is going to clone your state and return a new state for every action it sees. In your particular case, that might be fine, but if this reducer was part of a larger store, there would be lots of actions for which this reducer isn't changing anything.

A better practice would be to have each individual case statement return a newState, and if we don't hit any of the case statements just return the unchanged `state`